client/components/common/BuildList.tsx

```diff
@@ -96,6 +117,51 @@ const BuildList: React.FC<Props> = ({
     DEBOUNCE_INTERVAL,
   );
 
+  // Takes params from the router and filters them out so that the inherent username value is not in the address bar
+  // twice, also when values are undefined or empty strings. Returns a new object to use in formatQueryString.
+  const formatParams = (obj: ProfileQueryParams) =>
+    Object.entries(obj)
+      .filter(([, val]) => val !== undefined && val !== '' && val !== username)
+      .reduce((result, [key, val]) => {
+        const newResult = {
+          ...result,
+          [key]: val,
+        };
+        return newResult;
+      }, {} as ProfileQueryParams);
+
+  const formatQueryString = () => {
+    const params = formatParams({
+      ...router.query,
+      search,
+      class: classData?.classes.find((item) => item.id === dofusClassId)
+        ?.enName,
+      tags: tagIds
+        .map(
+          (tagId) =>
+            tagsData?.customSetTags.find((tag) => tagId === tag.id)?.enName,
+        )
+        .join(','),
+    });
+
+    return !Object.keys(params).length
```

Comment:
nit: remove the ! and reverse the ternary, or do Object.keys(params).length === 0 ...